### PR TITLE
Libhtp rs v13

### PR DESCRIPTION
**Pull request by catenacyber**
Link to [redmine](https://redmine.openinfosecfoundation.org/issues/2696) ticket: 2696

Describe changes:
- Use libhtp-rs.

Needed rebase from #11381 

Draft to see CI feedback with latest libhtp-rs changes, and not have too many conflicts

TODOs :
- get https://github.com/cccs-rtmorti/libhtp-rs/pull/15 merged and look into the remaining oss-fuzz reports for libhtp-rs
- look into QA results

### CODE REVIEW

File: rust/Cargo.toml.in [hunk 1]
```diff
@@ -7,7 +7,7 @@ edition = "2021"
 rust-version = "1.63.0"
 
 [workspace]
-members = [".", "./derive"]
+members = [".", "./derive", "./libhtp-rs"]
```
Comment (jasonish) [inline rust/Cargo.toml.in:10]: `libhtp-rs` seems a bit redundant now. Could it just be `libhtp`, or even `htp` to match the package name?

**Comment by suricata-qa** [comment]
Information:
 

ERROR: QA failed on SURI_TLPW2_single_suri_time.

ERROR: QA failed on SURI_TLPW2_single_alerts_cmp.

ERROR: QA failed on SURI_TLPW2_autofp_alerts_cmp.

ERROR: QA failed on SURI_TLPW1_suri_time.

ERROR: QA failed on SURI_TLPW1_files_sha256.

ERROR: QA failed on SURI_TLPR1_alerts_cmp.
 | field | baseline | test | % |
|--|--|--|--| 
| | SURI_TLPW2_single_stats_chk | | |
| .uptime | 776 | 813 | 104.77% |
| | SURI_TLPW1_stats_chk | | |
| .app_layer.error.http.parser | 47 | 2 | 4.26% |
| | SURI_TLPR1_stats_chk | | |
| .app_layer.error.http.parser | 700 | 369 | 52.71% | 

Pipeline 21383

File: rust/Cargo.toml.in (hunk 1 above)
Comment (catenacyber) [reply rust/Cargo.toml.in:10]: ok for htp

**Comment by catenacyber** [comment]
Rebased in https://github.com/OISF/suricata/pull/11460

**Pull request closed without merging**